### ISSUE TITLE
Typescript: Interfaces for CredentialProviderChain missing resolvePromise function

### ISSUE THREAD

**Issue by rhussmann** [aws/aws-sdk-js#1600]
**Version**: current on master

**Issue**
The [Typescript definition](https://github.com/aws/aws-sdk-js/blob/master/lib/credentials/credential_provider_chain.d.ts) doesn't define `CredentialProviderChain.resolvePromise`, even though it is defined in [code](https://github.com/aws/aws-sdk-js/blob/master/lib/credentials/credential_provider_chain.js#L162-L164) and in the [AWS JS SDK Documentation](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CredentialProviderChain.html#resolvePromise-property)

**Reference**
- [current type definition](https://github.com/aws/aws-sdk-js/blob/master/lib/credentials/credential_provider_chain.d.ts)
- [curent implementation](https://github.com/aws/aws-sdk-js/blob/master/lib/credentials/credential_provider_chain.js)
- [AWS JS SDK docs](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CredentialProviderChain.html)

**Requested Change**
Please regenerate the type definition for `lib/credentials/credential_provider_chain.d.ts` to include this method.

**Comment by miikaah** [comment]
Ran into this as well. Also, the constructor's providers argument could be optional.

Right now I have to call CredentialProviderChain like this which is pretty ugly:

```javascript
new AWS.CredentialProviderChain(undefined)['resolvePromise']();
```

**Comment by jeskew** [comment]
This was resolved in #1655.
